### ISSUE TITLE
5.0.0 - [Firebase/Firestore][I-FST000001]  Firebase Timestamp objects instead of as system Date objects

### ISSUE THREAD

**Issue by LiveRock** [flutter/flutter#18159]
Here is the error/warning. The solution give is for ioS Swift but I am writing in Flutter/Dart.
```
5.0.0 - [Firebase/Firestore][I-FST000001] The behavior for system Date objects stored in Firestore is going to change AND YOUR APP MAY BREAK.
To hide this warning and ensure your app does not break, you need to add the following code to your app before calling any other Cloud Firestore methods:
let db = Firestore.firestore()
let settings = db.settings

With this change, timestamps stored in Cloud Firestore will be read back as Firebase Timestamp objects instead of as system Date objects. So you will also need to update code expecting a Date to instead expect a Timestamp. For example:
// old:
let date: Date = documentSnapshot.get("created_at") as! Date
// new:
let timestamp: Timestamp = documentSnapshot.get("created_at") as! Timestamp
let date: Date = timestamp.dateValue()
Please audit all existing usages of Date when you enable the new behavior. In a future release, the behavior will be changed to the new behavior, so if you do<…>
```
I can't find settings in the below:
final Firestore db = Firestore.instance;
  // var settings = db.app.settings ?

please help

**Comment by MStanoevich** [comment]
Same issue, though on Android 

```
W/Firestore( 7877): (0.6.6-dev) [Firestore]: The behavior for java.util.Date objects stored in Firestore is going to change AND YOUR APP MAY BREAK.
W/Firestore( 7877): To hide this warning and ensure your app does not break, you need to add the following code to your app before calling any other Cloud Firestore methods:
W/Firestore( 7877):
W/Firestore( 7877): FirebaseFirestore firestore = FirebaseFirestore.getInstance();
W/Firestore( 7877): FirebaseFirestoreSettings settings = new FirebaseFirestoreSettings.Builder()
W/Firestore( 7877):     .setTimestampsInSnapshotsEnabled(true)
W/Firestore( 7877):     .build();
W/Firestore( 7877): firestore.setFirestoreSettings(settings);
W/Firestore( 7877):
W/Firestore( 7877): With this change, timestamps stored in Cloud Firestore will be read back as com.google.firebase.Timestamp objects instead of as system java.util.Date objects. So you will also need to update code expecting a java.util.Date to instead expect a Timestamp. For example:
W/Firestore( 7877):
W/Firestore( 7877): // Old:
W/Firestore( 7877): java.util.Date date = snapshot.getDate("created_at");
W/Firestore( 7877): // New:
W/Firestore( 7877): Timestamp timestamp = snapshot.getTimestamp("created_at");
W/Firestore( 7877): java.util.Date date = timestamp.toDate();
W/Firestore( 7877):
W/Firestore( 7877): Please audit all existing usages of java.util.Date when you enable the new behavior. In a future release, the behavior will be changed to the new behavior, so if you do not follow these steps, YOUR APP MAY BREAK.
```

Any progress on resolving it?

**Comment by MStanoevich** [comment]
Not sure if I should post this here but as my original issue is closed due to being pretty much the same as this one I am posting it here :P 

Originally I thought that this was the reason why my app kept crashing after uploading an Image to Firebase Storage but now I seem to get the following message as well:
```
Shutting down VM
E/AndroidRuntime(29999): FATAL EXCEPTION: main
E/AndroidRuntime(29999): Process: com.example.lnotes, PID: 29999
E/AndroidRuntime(29999): java.lang.NoSuchMethodError: No virtual method getDownloadUrl()Landroid/net/Uri; in class Lcom/google/firebase/storage/UploadTask$TaskSnapshot; or its super classes (declaration of 'com.google.firebase.storage.UploadTask$TaskSnapshot' appears in /data/app/com.example.exampleApp-1/base.apk:classes2.dex)
```

**Comment by zoechi** [comment]
@MStanoevich 

> my original issue is closed

A link might be helpful if you think it's related.

**Comment by MStanoevich** [comment]
True, true
https://github.com/flutter/flutter/issues/19356

**Comment by Asoseil** [comment]
i have a warning only on android

**Comment by 1and1get2** [comment]
my take is that flutter team will handle this transaction with new coming updates and we don't need to take any measurements ourselves right?

**Comment by osamagamal65** [comment]
any updates yet?

**Comment by gagaJay** [comment]
same issue here, any fix?

**Comment by 5ummit** [comment]
Been an issue for sometime now, any fixes?

**Comment by satish754ss** [comment]
Having the same problem,
With this change, timestamps stored in Cloud Firestore will be read back as com.google.firebase.Timestamp objects instead of as system java.util.Dat
e objects. So you will also need to update code expecting a java.util.Date to instead expect a Timestamp. For example:

**Comment by Stone2517** [comment]
@satish754ss 
Can you please give us your solution code.

**Comment by Stone2517** [comment]
When I place the Code it's getting an error. Where should the code?
please, someone, give me an example with code.

**Comment by zoechi** [comment]
Waiting for PR #765

**Comment by DanTup** [comment]
@zoechi The referenced PR was merged on 5 Dec 2015, was this supposed to link to another repo?

**Comment by zoechi** [comment]
@DanTup thanks for the hint! I forgot to prefix with `flutter/plugins` - fixed.

**Comment by Stone2517** [comment]
@zoechi
Problem is fixed or not ?

**Comment by satish754ss** [comment]
no its trying to fix this issue.

**Comment by mdanics** [comment]
[PR](https://github.com/flutter/plugins/pull/765) was merged 8 hours ago, does this fix the issue?

**Comment by kroikie** [comment]
Thank you for this issue, it should be resolved in version `0.8.2`.

**Comment by Stone2517** [comment]
@kroikie 
I'm getting an error.
`PlatformException(error, Attempt to invoke virtual method 'void com.google.firebase.auth.internal.zzab.zzf(int)' on a null object reference, null)`

**Comment by mi6crazyheart** [comment]
@kroikie One confusion, the example link which you've shared, I can see how to use that setting. But, at plugin Usage section (here -> https://github.com/flutter/plugins/tree/master/packages/cloud_firestore ) there is nowhere it has mentioned using some kind of "setting" thing.

**Comment by satish754ss** [comment]
settings method is not available for class Firestore.
 await firestore.settings(timestampsInSnapshotsEnabled: true); showing error in settings.

**Comment by kroikie** [comment]
@mi6crazyheart the README needs to be updated to show the new usage.

**Comment by mi6crazyheart** [comment]
@kroikie Thanks for the reply & clearing out my confusion.

**Comment by mijogu** [comment]
Has anyone been able to get this resolved? Not seeing any updates in the package readme. I am unable to find any references to settings.

**Comment by satish754ss** [comment]
Ya new update is available.

**Comment by satish754ss** [comment]
await firestore.settings(timestampsInSnapshotsEnabled: true);
Updated .yaml,  and add this package, cloud_firestore: ^0.8.2+1.

**Comment by mi6crazyheart** [comment]
> Has anyone been able to get this resolved? Not seeing any updates in the package readme. I am unable to find any references to settings.

Check this - https://github.com/flutter/plugins/blob/master/packages/cloud_firestore/example/lib/main.dart#L22

**Comment by satish754ss** [comment]
![screenshot 47](https://user-images.githubusercontent.com/36834824/47608251-4c747080-da48-11e8-8e00-91e061e3e56d.png)

**Comment by rlee1990** [comment]
Does anyone receive an error after the update when trying to sort by comparing timestamps?

**Comment by bankthaworn** [comment]
> Does anyone receive an error after the update when trying to sort by comparing timestamps?

Yeah! I still receive an error.

**Comment by kroikie** [comment]
@LiveRock

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/810. Any further collaboration will be done there.

**Comment by uV3301** [comment]
> Thank you for this issue, it should be resolved in version `0.8.2`.
> 
> See [example](https://github.com/flutter/plugins/blob/master/packages/cloud_firestore/example/lib/main.dart#L22) of how to use settings.

Can you update the page or check the link you shared. It shows 404 error and I still can't find any relevant solution to this.
